README.md

```diff
@@ -12,30 +15,107 @@ is simpler and easier to work with. Additionally, for developers familar with
 the Go programming language will be instantly comfortable with Gunk files and
 syntax.
 
+### Contents
+
+* [How it works](#how-it-works)
+* [Features](#features)
+* [Installing](#installing)
+* [Usage](#usage)
+* [Contributing](#contributing)
+
+## How it works
+
+Gunk works by using Go's standard library `go/ast` package (and related
+packages) to parse all `*.gunk` files in a "Gunk Package" (and applying the
+same kinds of rules as Go package), building the appropos protobuf messages. In
+turn, those are passed to `protoc-gen-*` tools, along with any passed options.
```

Comment:
I think this shows a bit too much internal detail to be this far up in the README. Users will spend 30s skimming the readme to read what this is, why they should use it, and how to install it - the implementation details are less important.

So I'd just move it at least below "usage".